sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statements.scala

```diff
@@ -475,3 +475,18 @@ case class ShowCurrentNamespaceStatement() extends ParsedStatement
 case class ShowTablePropertiesStatement(
     tableName: Seq[String],
     propertyKey: Option[String]) extends ParsedStatement
+
+/**
+ *  SHOW FUNCTIONS statement, as parsed from SQL
+ */
+case class ShowFunctionsStatement(
+    userScope: Boolean,
+    systemScope: Boolean,
+    pattern: Option[String],
+    functionName: Option[Seq[String]]) extends ParsedStatement
+
+object ShowFunctionsStatement{
+  val ALL = "all"
```

Comment:
Since these constants are used only in `visitShowFunctions`, does it make sense to define these here? I think the original implementation is also fine?

Reply:
We can inline them in the parser rule `visitShowFunctions`